test/strategies/elkilpstrategy.js

```diff
@@ -0,0 +1,308 @@
+            // The strategy contract should have correct balance on the staking contract
+            expect((await stakingContract.balanceOf(elkIlpStrategyV5.address)).toString()).to.equal(amount)
+
+            // We check if the totalDeposit has grown after our deposit
+            expect(BigNumber.from(totalDepositsAfter)).gt(BigNumber.from(totalDepositsBefore));
+        })
+    })
+
+    describe("Test withdraw functionality", async () => {
+        beforeEach(async () => {
+            await wavaxTokenContract.approve(elkRouterContract.address, BigNumber.from("1000000000000000000000000"));
+            await elkTokenContract.approve(elkRouterContract.address, BigNumber.from("1000000000000000000000000"));
+            await elkPairContract.approve(elkIlpStrategyV5.address, BigNumber.from("1000000000000000000000000"));
+
+            await elkRouterContract.swapExactTokensForTokens(BigNumber.from("10000000000000000"), BigNumber.from("5000000000000000"), [wavaxTokenContract.address, elkTokenContract.address], owner.address, 1807909162115)
+            await elkRouterContract.addLiquidity(wavaxTokenContract.address, elkTokenContract.address, await wavaxTokenContract.balanceOf(owner.address), await elkTokenContract.balanceOf(owner.address), 0, 0, owner.address, 1807909162115)
+        })
+
+        it('Withdraw correctly our ELP tokens from the Strategy', async () => {
+            // We get the amount of ELK token we have
+            const amount = (await elkPairContract.balanceOf(owner.address)).toString();
+            // We get the 'totalDeposit' before we deposit into the strategy
+            const totalDepositsBefore = (await elkIlpStrategyV5.totalDeposits()).toString();
+
+            // We deposit into the staking contract our ELP tokens
+            await elkIlpStrategyV5.deposit(BigNumber.from((await elkPairContract.balanceOf(owner.address)).toString()))
+            // We get the 'totalDeposit' after our deposit into the strategy
+            const totalDepositsAfter = (await elkIlpStrategyV5.totalDeposits()).toString();
+
+            // We make sure all balances are correct .
+            expect((await elkIlpStrategyV5.balanceOf(owner.address)).toString()).to.equal(amount)
+            expect((await stakingContract.balanceOf(elkIlpStrategyV5.address)).toString()).to.equal(amount)
+            expect(BigNumber.from(totalDepositsAfter)).gt(BigNumber.from(totalDepositsBefore));
+
+            // We withdraw from the strategy all tokens we deposited
+            await elkIlpStrategyV5.withdraw(BigNumber.from((await elkIlpStrategyV5.balanceOf(owner.address)).toString()))
+
+            // We check if balances are correct
+            expect((await elkIlpStrategyV5.balanceOf(owner.address)).toString()).to.equal("0")
+            expect((await stakingContract.balanceOf(elkIlpStrategyV5.address)).toString()).to.equal("0")
+            expect(BigNumber.from(totalDepositsAfter)).gt(BigNumber.from(totalDepositsBefore));
```

Comment:
You haven't updated totalDepositsAfter anywhere after checking it on https://github.com/yieldyak/smart-contracts/pull/24/files#diff-16bb4608c77d5c5d77085450f2040e54d9e3854bb46d4c86e67afc3df5b923a8R163 so it feels you either forgot to update it again, or forgot to delete this line

Reply:
Done